### PR TITLE
JAVA-3400

### PR DESCRIPTION
**Pull request by jyemin**
https://jira.mongodb.org/browse/JAVA-3400

Patch build: https://evergreen.mongodb.com/version/5d66d9ae61837d20056e0c5b

### CODE REVIEW

**Comment by jyemin** [comment]
Patch build is occasionally failing, albeit with a different error.  I'd still appreciate a look at the fix while I try to figure out what's wrong

**Comment by jyemin** [comment]
This looks like the root cause of the occasional patch failure: https://jira.mongodb.org/browse/SERVER-43079

File: driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java [hunk 1]
```diff
@@ -265,6 +265,10 @@ public ClusterDescription getCurrentDescription() {
         return description;
     }
 
+    private synchronized void updatePhase() {
+        phase.getAndSet(new CountDownLatch(1)).countDown();
```
Comment (rozza) [inline driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java]: Can this method also be used in `updateDescription`?

Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java]: Yes, will fix and merge.

**Pull request closed without merging**